tests/integration/test_manifest_submission.py

```diff
@@ -0,0 +1,97 @@
+    if not os.path.exists(temporary_manifest_storage):
+        os.makedirs(temporary_manifest_storage)
+    full_parent_path = os.path.abspath(temporary_manifest_storage)
+
+    download_location = create_temp_folder(full_parent_path)
+    yield download_location
+
+    # Cleanup after tests have used the temp folder
+    if os.path.exists(download_location):
+        shutil.rmtree(download_location)
+
+
+class TestManifestGeneration:
+    def test_submit_record_based_test_manifest_file_only(
+        self, helpers, syn_token, download_location, schedule_for_cleanup
+    ) -> None:
+        url = "http://localhost:3001/v1/model/submit"
+        data_type = "Biospecimen"
+        params = {
+            "schema_url": "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld",
+            "data_model_labels": "class_label",
+            "data_type": data_type,
+            "dataset_id": "syn63561474",
+            "manifest_record_type": "file_only",
+            "restrict_rules": "false",
+            "hide_blanks": "false",
+            "asset_view": "syn63561606",
+            "table_manipulation": "replace",
+            "table_column_names": "class_label",
+            "annotation_keys": "class_label",
+            "file_annotations_upload": "false",
+        }
+
+        syn = Synapse()
+        syn.login(authToken=syn_token, silent=True)
+
+        headers = {"Authorization": f"Bearer {syn_token}"}
+        test_manifest_path = helpers.get_data_path(
+            "mock_manifests/mock_example_biospecimen_manifest.csv"
+        )
+
+        # THEN we expect a successful response
+        response = requests.post(
+            url,
+            headers=headers,
+            params=params,
+            files={"file_name": open(test_manifest_path, "rb")},
+        )
+        assert response.status_code == 200
+
+        # Get the manifest ID from the response
+        manifest_id = response.json()
+
+        # Load then manifest from synapse
+        manifest_data = syn.get(
+            manifest_id,
+            downloadLocation=download_location,
+            ifcollision="overwrite.local",
+        )
+        # make sure that the file name of manifest is correct
+        assert (
+            manifest_data["properties"]["name"]
+            == (f"synapse_storage_manifest_{data_type}.csv").lower()
+        )
+
+        # make sure that entity id and id columns were added
+        manifest_file_path = os.path.join(
+            download_location, manifest_data["properties"]["name"]
+        )
+        manifest_submitted_df = pd.read_csv(manifest_file_path)
+        assert "entityId" in manifest_submitted_df.columns
+        assert "Id" in manifest_submitted_df.columns
+
+        # make sure that Id column is not empty
+        assert manifest_submitted_df["Id"].notnull().all()
+
+        # clean up
+        schedule_for_cleanup(CleanupItem(manifest_id))
```

Comment:
@linglp 
Can you add the cleanup directly after getting the ID of the resource. The way this is setup now is that if the assertion fails this line won't be hit, nor the resource cleaned up.

Reply:
Yes of course.